src/AutoMerge/guidelines.jl

```diff
@@ -54,11 +66,45 @@ function meets_compat_for_julia(working_directory::AbstractString, pkg, version)
     return false, "There is no compat entry for `julia`."
 end
 
-const guideline_compat_for_all_deps =
-    Guideline("Compat (with upper bound) for all dependencies",
-              data -> meets_compat_for_all_deps(data.registry_head,
-                                                data.pkg,
-                                                data.version))
+const guideline_compat_for_all_deps = Guideline(;
+    info = "Compat (with upper bound) for all dependencies",
+    docs = """
+    Dependencies: All dependencies should have `[compat]` entries that are upper bounded and only include a finite number of breaking releases.
+    For example, the following `[compat]` entries meet the criteria for automatic merging:
+    ```toml
+    [compat]
+    PackageA = "1"          # [1.0.0, 2.0.0), has upper bound (good)
+    PackageB = "0.1, 0.2"   # [0.1.0, 0.3.0), has upper bound (good)
+    ```
+    The following `[compat]` entries do NOT meet the criteria for automatic merging:
+    ```toml
+    [compat]
+    PackageC = ">=3"        # [3.0.0, ∞), no upper bound (bad)
+    PackageD = ">=0.4, <1"  # [-∞, ∞), no lower bound, no upper bound (very bad)
+    ```
+    Please note: each `[compat]` entry must include only a finite number of breaking releases. Therefore, the following `[compat]` entries do NOT meet the criteria for automatic merging:
+    ```toml
+    [compat]
+    PackageE = "0"          # includes infinitely many breaking 0.x releases of PackageE (bad)
+    PackageF = "0.2 - 0"    # includes infinitely many breaking 0.x releases of PackageF (bad)
+    PackageG = "0.2 - 1"    # includes infinitely many breaking 0.x releases of PackageG (bad)
+    ```
+    See [Pkg's documentation][pkg-compat] for specification of `[compat]` entries in your
+    `Project.toml` file.
+
+    (**Note:** Standard libraries are excluded for this criterion since they are bundled
+    with Julia, and, hence, implicitly included in the `[compat]` entry for Julia.
+    For the time being, JLL dependencies are also excluded for this criterion because they
+    often have non-standard version numbering schemes; however, this may change in the future.)
+
+    You may find [CompatHelper.jl](https://github.com/bcbi/CompatHelper.jl) helpful for maintaining up-to-date `[compat]` entries.
```

Comment:
Possibly it could also be helpful to point to my `PackageCompatUI` package..

Reply:
Done.